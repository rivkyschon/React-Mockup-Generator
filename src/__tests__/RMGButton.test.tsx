// src/components/__tests__/RMGButton.test.tsx

import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import RMGButton from '../components/core/RMGButton';

describe('RMGButton', () => {
  it('renders correctly', () => {
    const {asFragment} = render(<RMGButton title="Click Me" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the correct title', () => {
    const {getByText} = render(<RMGButton title="Click Me" />);
    // expect(getByText('Click Me')).toBeInTheDocument();
  });

  it('handles click events', () => {
    const handleClick = jest.fn();
    const {getByText} = render(
      <RMGButton title="Click Me" onClick={handleClick} />,
    );
    fireEvent.click(getByText('Click Me'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
