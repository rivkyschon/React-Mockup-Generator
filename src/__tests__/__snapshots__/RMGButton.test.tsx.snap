// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RMGButton renders correctly 1`] = `
<DocumentFragment>
  <button
    class="bg-black text-white py-2 px-4 rounded hover:bg-opacity-90 focus:outline-none"
  >
    Click Me
  </button>
</DocumentFragment>
`;
